src/os/bluestore/BlueStore.cc

```diff
@@ -8729,6 +9824,8 @@ int BlueStore::_fsck_on_open(BlueStore::FSCKDepth depth, bool repair)
     }
 
     dout(1) << __func__ << " checking freelist vs allocated" << dendl;
+    // skip freelist vs allocated compare when we have Null fm
+    if (!fm->is_null_manager())
```

Comment:
You might want to modify this verification by comparing allocator's allocated extents vs. list of actual allocations from onodes, bluefs etc. And repair if needed by updating the allocator. Hence no need to bypass the testing stuff in store_test

Reply:
Do you mean build an allocation-map from Onodes and then compare it with the allocation-map read from file?
It is easy to add this test as I already got the code (invoked from bluestore-tool), but can be time consuming (minutes ...).
If you think it is worth the time I can add it

As for fixing - once we find the first error I can set a flag to copy the allocator generated from Onodes to the shared-allocator (will take few extra seconds )

Reply:
yeah, that's what I am suggesting. Actually such a check wouldn't take much time - fsck reads all the objects any way. If one uses bitmap to track free/allocated extents it should be fast enough...
May be done later via next PR though..